ctapipe/reco/HillasReconstructor.py

```diff
@@ -129,20 +133,27 @@ class are set to np.nan
         ------
         TooFewTelescopesException
             if len(hillas_dict) < 2
+        InvalidWidthException
+            if any width is np.nan or 0
         '''
 
         # filter warnings for missing obs time. this is needed because MC data has no obs time
         warnings.filterwarnings(action='ignore', category=MissingFrameAttributeWarning)
 
-        # stereoscopy needs at least two telescopes with valid widths
-        valid_telescopes = sum([1 if not np.isnan(hillas_dict[x]['width'].value)
-                                else 0
-                                for x in hillas_dict])
-
-        if valid_telescopes < 2:
+        # stereoscopy needs at least two telescopes
+        if len(hillas_dict) < 2:
             raise TooFewTelescopesException(
                 "need at least two telescopes, have {}"
-                .format(valid_telescopes))
+                .format(len(hillas_dict)))
+
+        # check for np.nan or 0 width's as these screw up weights
+        if any([np.isnan(hillas_dict[tel]['width'].value) for tel in hillas_dict]):
```

Comment:
Better in one loop and use a generator expression?

```
any(np.isnan(h.width.value) or h.width.value == 0 for h in hillas_dict.values())
```

Reply:
Fine with me. I wanted to seperate the error messages but we can combine them if you feel like we dont gain anything with it

Reply:
I would prefer to print two different messages in case of `w=0` or `w=NaN`.
Ok to use one generator expression per each case.